src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -361,43 +360,37 @@ private enum ParamKind
             // Task<_> -> Int32
             private static TypeSymbol TranslateReturnType(CSharpCompilation compilation, TypeSymbol returnType)
             {
-                if (returnType.IsGenericTaskType(compilation)) {
-
-                    return compilation.GetSpecialType(SpecialType.System_Int32);
-                }
-                else
-                {
-                    return compilation.GetSpecialType(SpecialType.System_Void);
-                }
+                return returnType.IsGenericTaskType(compilation)
+                    ? compilation.GetSpecialType(SpecialType.System_Int32)
+                    : compilation.GetSpecialType(SpecialType.System_Void);
```

Comment:
We *should* be protected at this point, as this code is only executed if an async-main entrypoint was found, which can't happen if GetSpecialType is broken.